src/engine/strat_engine/liminal/device_info.rs

```diff
@@ -669,17 +669,22 @@ impl DeviceSet {
     /// The encryption information and devices registered for stopped pools to
     /// be exported over the API.
     ///
-    /// Error from gather_encryption_info is converted into an option because
-    /// unlocked Stratis devices and LUKS2 devices on which the Stratis devices are
-    /// stored may appear at different times in udev. This is not necessarily
-    /// an error case and may resolve itself after more devices appear in udev.
-    pub fn stopped_pool_info(&self) -> Option<StoppedPoolInfo> {
-        gather_encryption_info(
+    /// Converts an error result from gather_encryption_info into maximum
+    /// uncertainty, since the error means that encryption information is
+    /// completely missing from one device.
+    pub fn stopped_pool_info(&self) -> StoppedPoolInfo {
+        let info = gather_encryption_info(
             self.internal.len(),
             self.internal.values().map(|info| info.encryption_info()),
         )
-        .ok()
-        .map(|info| StoppedPoolInfo {
+        .unwrap_or({
```

Comment:
I don't think this does the right thing. Errors are only returned when they're a mixture of encrypted and unencrypted devices. I think this aspect of the code should be handled in `engine/shared.rs` in the gather methods. Gather takes a function as a parameter so it can likely be modified to calculate whether it's `PoolEncryption::KeyDesc(MaybeInconsistent::Yes)`, etc.

Reply:
When the two arguments to gather_encryption_info are constructed, the iterator derived from internal.values() returns ```None``` in one place, so that the flattened length is less than the unflattened.

I think that what you're saying is that when the information is being gathered, it is possible to determine what is the encryption info on the devices w/ valid encryption info.

So, if all the devices w/ valid LUKS metadata are KeyDescription only, then you would like to see ```PoolEncryptionInfo::KeyDesc(MaybeInconsistent::Yes)``` rather than both showing inconsistent?

I think we can do that, but my own position is that in the general case where the LUKS info is actually missing because invalid, that information is not certainly known, so I decided to use the most confused value.

If all the devices had valid LUKS2 metadata, and each had a key description, but some key descriptions were different from others, then I would think that it would be appropriate to return
`PoolEncryptionInfo::KeyDesc(MaybeInconsistent::Yes)```,  but it actually make more sent to me to return ```Both``` as I do here, in this situation.